extensions/azure-storage-datalake/deployment/src/main/java/org/apache/camel/quarkus/component/azure/storage/datalake/deployment/AzureStorageDatalakeProcessor.java

```diff
@@ -54,10 +55,20 @@ ExtensionSslNativeSupportBuildItem activateSslNativeSupport() {
     ReflectiveClassBuildItem registerForReflection(CombinedIndexBuildItem combinedIndex) {
         IndexView index = combinedIndex.getIndex();
 
-        List<String> dtos = new LinkedList<>(index.getKnownClasses().stream()
+        LinkedHashSet<String> dtos = new LinkedHashSet<>(index.getKnownClasses().stream()
                 .map(ci -> ci.name().toString())
                 .filter(n -> n.startsWith("com.azure.storage.file.datalake.implementation.models"))
-                .collect(Collectors.toList()));
+                .toList());
+
+        dtos.addAll(index.getAllKnownImplementations(HttpResponseException.class).stream()
+                .map(ci -> ci.name().toString())
+                .toList());
+        dtos.addAll(index.getAllKnownImplementations(JsonSerializable.class).stream()
+                .map(ci -> ci.name().toString())
+                .toList());
+        dtos.addAll(index.getAllKnownImplementations(XmlSerializable.class).stream()
+                .map(ci -> ci.name().toString())
+                .toList());
```

Comment:
`HttpResponseException` is a concrete class and not an interface. So the Jandex lookup probably returns nothing. Reflection for that class is already covered in `camel-quarkus-azure-support-core` anyway, so you can probably remove the code from the datalake extension.

I think we should move reflection config for `JsonSerializable` and `XmlSerializable ` into `camel-quarkus-azure-support-core` and clean up any other extensions that try to set up reflection for it.

Reply:
moved